drivers/sensor/nxp/nxp_mcx_temp/temp_mcx.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_mcx_temperature
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/devicetree.h>
+
+LOG_MODULE_REGISTER(temp_mcx, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Four ADC samples: first two are dropped, then VBE1 and VBE8 */
+#define TEMP_MCX_ADC_SAMPLES 1
+
+struct temp_mcx_config {
+	const struct device *adc;
+	struct adc_sequence adc_seq;
+	struct adc_channel_cfg ch_cfg;
+};
+
+struct temp_mcx_data {
+	float temperature;
+	uint16_t buffer[TEMP_MCX_ADC_SAMPLES];
+};
+
+static int temp_mcx_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct temp_mcx_config *config = dev->config;
+	struct temp_mcx_data *data = dev->data;
+	int err;
+	uint16_t vbe1 = 0U;
+	uint16_t vbe8 = 0U;
+	uint32_t conv_result_shift = 0U;
```

Comment:
nit, declare variables in reverse christmas tree order , i.e. longest line to shortest line. This is the de facto standard style in zephyr.